### PR TITLE
Don't fire update event when merging values are equal

### PR DESCRIPTION
**Pull request by ahmetmircik**
closes https://github.com/hazelcast/hazelcast/issues/12175

### CODE REVIEW

**Comment by pveentjer** [comment]
Will there be a forward port or has the logic changed in 3.10?

File: hazelcast/src/main/java/com/hazelcast/cache/impl/record/CacheRecordFactory.java [hunk 1]
```diff
@@ -81,4 +81,21 @@ public static boolean isExpiredAt(long expirationTime, long now) {
         return expirationTime > -1 && expirationTime <= now;
     }
 
+    public boolean isEqual(Object existingValue, Object newValue) {
```
Comment (taburet) [inline hazelcast/src/main/java/com/hazelcast/cache/impl/record/CacheRecordFactory.java]: Can `dataExistingValue` and `objectExistingValue` be `null` in this method?

**Comment by mmedenjak** [comment]
@ahmetmircik can you rebase and address the requested change?

**Comment by ahmetmircik** [comment]
@mmedenjak please see this: https://github.com/hazelcast/hazelcast/pull/13791

**Pull request closed without merging**